packages/donations-app/app/components/dashboard/tabs/money/charityDashboardDonationsController.js

```diff
@@ -173,6 +174,28 @@ angular.module('aliceApp')
       });
     }
 
+    // Hacky function to calculate received value for each donation
+    // (here we use word "users" to be consistent with the rest of the code)
+    // users here mean donations - it could be misleading and we should refactor it later
+    function calculateReceivedForUsers(users) {
```

Comment:
Are you sure that donations are always in the same order? How about sorting them by donation time to be 100% sure.